apps/emqx_rule_engine/src/emqx_rule_events.erl

```diff
@@ -79,11 +80,24 @@ event_names() ->
         'delivery.dropped'
     ].
 
-%% for documentation purposes
 event_topics_enum() ->
     [
         '$events/sys/alarm_activated',
         '$events/sys/alarm_deactivated',
+        '$events/client/connected',
+        '$events/client/disconnected',
+        '$events/client/connack',
+        '$events/auth/check_authn_complete',
+        '$events/auth/check_authz_complete',
+        '$events/session/subscribed',
+        '$events/session/unsubscribed',
+        '$events/message/delivered',
+        '$events/message/acked',
+        '$events/message/dropped',
+        '$events/message/delivery_dropped',
+        '$events/message_transformation/failed',
+        '$events/schema_validation/failed',
+        %% Topics below are kept for backwards compatibility.
         '$events/client_connected',
```

Comment:
do we need to keep the atoms for backward compatibility? 
I wonder if we can translate the events when compiling the rule

Reply:
I think they are atoms just for this enum:

https://github.com/emqx/emqx/blob/08562c47a7c71d391526baba6d5ea8f38c1bac9d/apps/emqx_rule_engine/src/emqx_rule_api_schema.erl#L123-L125